cg/cli/archive.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+import click
+from click.core import ParameterSource
+
+from cg.constants.constants import DRY_RUN
+from cg.meta.archive.archive import SpringArchiveAPI
+from cg.models.cg_config import CGConfig
+
+LOG = logging.getLogger(__name__)
+DEFAULT_SPRING_ARCHIVE_COUNT = 200
+
+
+@click.group()
+@click.pass_obj
+def archive():
+    """Archive utilities."""
+    pass
+
+
+@archive.command("archive-spring-files")
+@click.option(
+    "-l",
+    "--limit",
+    help="Give a limit to the amount of files to archive.",
+    default=DEFAULT_SPRING_ARCHIVE_COUNT,
+    show_default=True,
+)
+@click.option(
+    "--archive-all",
+    help="Use in order to archive all non-archived-files",
+    is_flag=True,
+    default=False,
+    show_default=True,
+)
+@DRY_RUN
+@click.pass_obj
+def archive_spring_files(context: CGConfig, limit: int | None, archive_all: bool):
+    """Archives non-archived spring files.
+    Raises:
+        click.Abort if both a limit to the number of spring files to archive and archive_all is specified.
+    """
+
+    if (
+        click.get_current_context().get_parameter_source("limit") == ParameterSource.COMMANDLINE
+        and limit
+        and archive_all
+    ):
+        LOG.warning(
+            "Incorrect input parameters - please do not provide both a limit and set --archive-all to true."
+        )
+        raise click.Abort
```

Comment:
If possible, it would be nice to either have this logic set using the Click library above or do the validation in the logic below. I imagine this kind of logic can become pretty extensive, so would be nice to have a good pattern for it (which does not entail having a lot of logic here)

Reply:
Apparently, `Click` does support this.

```python
# You can define custom Click options that handle mutual exclusivity and default values
limit_option = click.option(
    "-l",
    "--limit",
    help="Limit the number of files to archive.",
    default=DEFAULT_SPRING_ARCHIVE_COUNT,
    show_default=True,
    cls=MutexOption,
    not_required_if=['archive_all'],
)

archive_all_option = click.option(
    "--archive-all",
    help="Archive all non-archived files.",
    is_flag=True,
    default=False,
    show_default=True,
    cls=MutexOption,
    not_required_if=['limit'],
)

```

Reply:
Love that!

Reply:
Seems like a chatgpt bluff 😢

Reply:
Yeah, at least you have to define the mutex class yourself, lets leave it